src/levanter/compat/hf_checkpoints.py

```diff
@@ -55,269 +135,498 @@ def to_dict(self) -> dict:
+                state_dict = torch.load(model_path, map_location=device)
+                state_dict = {k: v.cpu().numpy() for k, v in state_dict.items()}
+
+        return state_dict
+
+    def load_lm_model(
+        self,
+        lm_model_cls: Type[LmWithHfSerializationMixin],
+        ref: Optional[Union[str, RepoRef]] = None,
+        axis_mapping: Optional[ResourceMapping] = None,
+    ) -> LmWithHfSerializationMixin:
+        """
+        Loads a levanter model from a huggingface checkpoint.
+
+        Args:
+            lm_model_cls: The model class to load
+            ref: The reference to load from. If None, will use the reference_checkpoint
+            axis_mapping: The axis mapping to use for sharding. If None, will use the context axis mapping
+        """
+        state_dict = self.load_state_dict(ref)
+        hf_config = self.hf_config_from_hf_checkpoint(ref)
+        config = self.config_from_hf_config(hf_config)
+
+        vocab_size = hf_config.vocab_size
+
+        Vocab = self.Vocab.resize(vocab_size)
+
+        ignore_prefix: Optional[str] = None
+        if self.ignore_prefix:
+            for k in state_dict.keys():
+                if k.startswith(f"{self.ignore_prefix}."):
+                    ignore_prefix = self.ignore_prefix
+                    break
+
+        # TODO: i still think this isn't the best way to do this
+        with jax.default_device(jax.devices("cpu")[0]):
+            lev_model = filter_eval_shape(lm_model_cls.init, Vocab, config, key=PRNGKey(0))
+            lev_model = lev_model.from_state_dict(state_dict, prefix=ignore_prefix)
+
+        if axis_mapping is not None:
+            lev_model = haliax.shard_with_axis_mapping(lev_model, axis_mapping)
+        else:
+            lev_model = haliax.auto_sharded(lev_model)
+
+        return lev_model
+
+    def save_model_local(
+        self,
+        model: LmWithHfSerializationMixin,
+        path: str,
+        save_tokenizer: bool = True,
+        save_reference_code: bool = True,
+    ):
+        """
+        Saves a HF-compatible checkpoint to a local path.
+        :param model:
+        :param path:
+        :param save_tokenizer: Save the tokenizer to the checkpoint
+        :param save_reference_code: Save any code from the reference checkpoint
+        :return:
+        """
+        logger.info(f"Saving HF-compatible checkpoint to {path}")
+        os.makedirs(path, exist_ok=True)
+
+        # save code first because we'll likely be overwriting it
+        if save_reference_code:
+            self._save_code_local(path)
+
+        config = model.config.to_hf_config(model.Vocab.size)
+        dict_config = config.to_dict()
+
+        # copy over the default keys
+        for k in KEYS_TO_COPY_FROM_BASE_CONFIG:
+            attr = getattr(self.default_hf_config, k, None)
+            if attr is not None:
+                dict_config[k] = attr
+
+        if self.config_overrides:
+            dict_config.update(self.config_overrides)
```

Comment:
Instead of overriding config here, can we pass `config_overrides` in `to_hf_config()`?

Reply:
sure, i was actually intending on removing config_overrides from to_hf_config but happy to go the other way